receiver/k8sobjectsreceiver/config.go

```diff
@@ -128,7 +129,13 @@ func (c *Config) getValidObjects() (map[string][]*schema.GroupVersionResource, e
 
 	res, err := dc.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		// Check if Partial result is returned from discovery client, that means some API servers have issues,
+		// but we can still continue, as we check for the needed groups later in Validate function.
+		var errDiscovery *discovery.ErrGroupDiscoveryFailed
```

Comment:
Mind switching to using [this helper function](https://pkg.go.dev/k8s.io/client-go/discovery#IsGroupDiscoveryFailedError) instead?